packages/ringcentral-widgets/components/Dialog/index.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import classnames from 'classnames';
+
+import styles from './styles.scss';
+import i18n from './i18n';
+import Button from '../Button';
+import CloseIcon from '../../assets/images/CloseIcon.svg';
+
+function FlatButton({
+  className, disabled, onClick, children
+}) {
+  return (
+    <div
+      className={classnames(
+        className,
+        styles.flatBtn,
+        styles.text,
+        disabled && styles.disabled,
+      )}
+      onClick={!disabled && onClick}
+    >
+      {children}
+    </div>
+  );
+}
+FlatButton.propTypes = {
+  className: propTypes.string,
+  disabled: propTypes.bool,
+  onClick: propTypes.func,
+  children: propTypes.node,
+};
+
+FlatButton.defaultProps = {
+  className: undefined,
+  disabled: false,
+  onClick: undefined,
+  children: undefined,
+};
+
+export default function Dialog({
+  children,
+  title,
+  onConfirm,
+  onCancel,
+  textConfirm,
+  textCancel,
+  currentLocale,
+  className,
+  cancelBtnClassName,
+  confirmBtnClassName,
+  showTitle,
+  showCloseBtn,
+  headerClassName,
+  contentClassName,
+  footerClassName,
+}) {
+  const footer =
+    !currentLocale || (!onCancel && !onConfirm) ? null : (
+      <div className={classnames(styles.footer, footerClassName)}>
+        {onCancel ? (
+          <FlatButton
+            className={classnames(
+              styles.btn,
+              styles.cancelBtn,
+              cancelBtnClassName,
+            )}
+            onClick={onCancel}
+          >
+            {textCancel || i18n.getString('cancel', currentLocale)}
+          </FlatButton>
+        ) : null}
+        {onConfirm ? (
+          <FlatButton
+            className={classnames(
+              styles.btn,
+              styles.confirmBtn,
+              confirmBtnClassName,
+            )}
+            onClick={onConfirm}
+          >
+            {textConfirm || i18n.getString('confirm', currentLocale)}
+          </FlatButton>
+        ) : null}
+      </div>
+    );
+  return (
+    <div
+      className={classnames(styles.dialog, className)}
+    >
+      {showTitle ? (
+        <div className={classnames(styles.header, headerClassName)}>
```

Comment:
Could we make header more semantic, prepare for the a11y. I think use section and header is appropriate here.